cirq/protocols/channel.py

```diff
@@ -175,3 +178,19 @@ def has_channel(val: Any) -> bool:
 
     # No has methods, use `_channel_` or delegates instead.
     return channel(val, None) is not None
+
+
+def validate_channel(supports_channel: SupportsChannel, rtol: int = 1e-5, atol: int = 1e-8):
+    """Validates that the channel's unitaries are a valid representation."""
+    channel_result = channel(supports_channel, None)
+    if channel_result is None:
+        raise TypeError("object of type '{}' has no _channel_ or _mixture_ or "
+                        "_unitary_ method.".format(type(supports_channel)))
+    shapes = list(set([m.shape for m in channel_result]))
+    if len(shapes) != 1 or len(shapes[0]) != 2 or shapes[0][0] != shapes[0][1]:
```

Comment:
I know that currently we only support channels whose input space is equal to its output space, but technically this not required.  I would suggest not enforcing this.  There are plans for operators that would need to change space like #1452.